File: ts/bindings/__file_snapshots__/Person.py
class Person(Thing):
    """A person (alive, dead, undead, or fictional)."""

    address: Optional[str] = None
    """Postal address for the person."""

    affiliations: Optional[Array["Organization"]] = None
    """Organizations that the person is affiliated with."""

    emails: Optional[Array[str]] = None
    """Email addresses for the person."""

    familyNames: Optional[Array[str]] = None
    """Family name. In the U.S., the last name of a person."""

    funders: Optional[Array[Union["Organization", "Person"]]] = None
    """A person or organization that supports (sponsors) something through
some kind of financial contribution.
"""

    givenNames: Optional[Array[str]] = None
    """Given name. In the U.S., the first name of a person."""

    honorificPrefix: Optional[str] = None
    """An honorific prefix preceding a person's name such as Dr/Mrs/Mr."""

    honorificSuffix: Optional[str] = None
    """An honorific suffix after a person's name such as MD/PhD/MSCSW."""

    jobTitle: Optional[str] = None
    """The job title of the person (for example, Financial Manager)."""

    memberOf: Optional[Array["Organization"]] = None
    """An organization (or program membership) to which this person belongs."""

    telephoneNumbers: Optional[Array[str]] = None
    """Telephone numbers for the person."""


    def __init__(
        self,
        address: Optional[str] = None,
        affiliations: Optional[Array["Organization"]] = None,
        alternateNames: Optional[Array[str]] = None,
        description: Optional[Union[str, Array["Node"]]] = None,
        emails: Optional[Array[str]] = None,
        familyNames: Optional[Array[str]] = None,
        funders: Optional[Array[Union["Organization", "Person"]]] = None,
        givenNames: Optional[Array[str]] = None,
        honorificPrefix: Optional[str] = None,
        honorificSuffix: Optional[str] = None,
        id: Optional[str] = None,
        identifiers: Optional[Array[Union[str, "PropertyValue"]]] = None,
        jobTitle: Optional[str] = None,
        memberOf: Optional[Array["Organization"]] = None,
        meta: Optional[Dict[str, Any]] = None,
        name: Optional[str] = None,
        telephoneNumbers: Optional[Array[str]] = None,
        url: Optional[str] = None
    ) -> None:
        super().__init__(
            alternateNames=alternateNames,
            description=description,
            id=id,
            identifiers=identifiers,
            meta=meta,
            name=name,
            url=url
        )
        if address is not None:
            self.address = address
        if affiliations is not None:
            self.affiliations = affiliations
        if emails is not None:
            self.emails = emails
        if familyNames is not None:
            self.familyNames = familyNames
        if funders is not None:
            self.funders = funders
        if givenNames is not None:
            self.givenNames = givenNames
        if honorificPrefix is not None:
            self.honorificPrefix = honorificPrefix
        if honorificSuffix is not None:
            self.honorificSuffix = honorificSuffix
        if jobTitle is not None:
            self.jobTitle = jobTitle
        if memberOf is not None:
            self.memberOf = memberOf
        if telephoneNumbers is not None:
            self.telephoneNumbers = telephoneNumbers

File: ts/bindings/__file_snapshots__/Person.R
#' A person (alive, dead, undead, or fictional).
#'
#' @name Person
#' @param address Postal address for the person.
#' @param affiliations Organizations that the person is affiliated with.
#' @param alternateNames Alternate names (aliases) for the item.
#' @param description A description of the item.
#' @param emails Email addresses for the person.
#' @param familyNames Family name. In the U.S., the last name of a person.
#' @param funders A person or organization that supports (sponsors) something through some kind of financial contribution.
#' @param givenNames Given name. In the U.S., the first name of a person.
#' @param honorificPrefix An honorific prefix preceding a person's name such as Dr/Mrs/Mr.
#' @param honorificSuffix An honorific suffix after a person's name such as MD/PhD/MSCSW.
#' @param id The identifier for this item.
#' @param identifiers Any kind of identifier for any kind of Thing.
#' @param images Images of the item.
#' @param jobTitle The job title of the person (for example, Financial Manager).
#' @param memberOf An organization (or program membership) to which this person belongs.
#' @param meta Metadata associated with this item.
#' @param name The name of the item.
#' @param telephoneNumbers Telephone numbers for the person.
#' @param url The URL of the item.
#' @seealso \code{\link{Thing}}
#' @export
Person <- function(
  address,
  affiliations,
  alternateNames,
  description,
  emails,
  familyNames,
  funders,
  givenNames,
  honorificPrefix,
  honorificSuffix,
  id,
  identifiers,
  images,
  jobTitle,
  memberOf,
  meta,
  name,
  telephoneNumbers,
  url
){
  self <- Thing(
    alternateNames = alternateNames,
    description = description,
    id = id,
    identifiers = identifiers,
    images = images,
    meta = meta,
    name = name,
    url = url
  )
  self$type <- as_scalar("Person")
  self[["address"]] <- check_property("Person", "address", FALSE, missing(address), "character", address)
  self[["affiliations"]] <- check_property("Person", "affiliations", FALSE, missing(affiliations), Array(Organization), affiliations)
  self[["emails"]] <- check_property("Person", "emails", FALSE, missing(emails), Array("character"), emails)
  self[["familyNames"]] <- check_property("Person", "familyNames", FALSE, missing(familyNames), Array("character"), familyNames)
  self[["funders"]] <- check_property("Person", "funders", FALSE, missing(funders), Array(Union(Organization, Person)), funders)
  self[["givenNames"]] <- check_property("Person", "givenNames", FALSE, missing(givenNames), Array("character"), givenNames)
  self[["honorificPrefix"]] <- check_property("Person", "honorificPrefix", FALSE, missing(honorificPrefix), "character", honorificPrefix)
  self[["honorificSuffix"]] <- check_property("Person", "honorificSuffix", FALSE, missing(honorificSuffix), "character", honorificSuffix)
  self[["jobTitle"]] <- check_property("Person", "jobTitle", FALSE, missing(jobTitle), "character", jobTitle)
  self[["memberOf"]] <- check_property("Person", "memberOf", FALSE, missing(memberOf), Array(Organization), memberOf)
  self[["telephoneNumbers"]] <- check_property("Person", "telephoneNumbers", FALSE, missing(telephoneNumbers), Array("character"), telephoneNumbers)
  class(self) <- c(class(self), "Person")
  self
}

